var/spack/repos/builtin/packages/ImageMagick/package.py

```diff
@@ -42,6 +43,9 @@ class Imagemagick(Package):
     depends_on('libtiff')
     depends_on('ghostscript')
 
+    def url_for_version(self, version):
+        return "https://github.com/ImageMagick/ImageMagick/archive/{0}.tar.gz".format(version)
```

Comment:
>
> They only have the bleeding edge of the current release (7.0.2-7) and the
> previous release (6.9.5-5). There was a recent security vulnerability
> discovered <https://imagetragick.com/>, so that may also be why.


I see:
     http://www.imagemagick.org/download/

Unless someone complains, let's just use the versions they provide there.
Old versions were probably removed for a reason; please do not include
versions with known security vulnerabilities.

They make it clear that the authoritative Git repo for ImageMagick is found
at:
    http://git.imagemagick.org/repos/ImageMagick

If we need to use a git repo, I think we should pull from this one, not
GitHub or GitLab.

-- Elizabeth


On Thu, Aug 11, 2016 at 3:15 PM, Adam J. Stewart <notifications@github.com>
wrote:

> In var/spack/repos/builtin/packages/ImageMagick/package.py:
>
> > @@ -42,6 +43,9 @@ class Imagemagick(Package):
> >      depends_on('libtiff')
> >      depends_on('ghostscript')
> >
> > +    def url_for_version(self, version):
> > +        return "https://github.com/ImageMagick/ImageMagick/archive/{0}.tar.gz".format(version)
>
> This should work automatically, but the version was getting mangled. I
> think it must be due to the dash in the version. Adding the url_for_version
> solves the problem.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LLNL/spack/pull/1041/files/f147d645272bc96e57c707f39a80b64d1e637fb1..633e540a1718a5cc515725b13d3f1740bb950bb6#r74485252>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB1cdwGBvwXNLcLznUDcGMxC487b4UKQks5qe3TQgaJpZM4IzSyf>
> .
>